App/navigators/RootStack.tsx

```diff
@@ -1,36 +1,96 @@
+import { useNavigation } from '@react-navigation/core'
 import { createStackNavigator } from '@react-navigation/stack'
-import React, { useState } from 'react'
+import React, { useContext, useState } from 'react'
+import { Text, TouchableOpacity, View } from 'react-native'
 
+import { DispatchAction } from '../Reducer'
+import { Context } from '../Store'
+import { Colors } from '../Theme'
+import Arrow from '../assets/img/large-arrow.svg'
+import { Screens } from '../constants'
+import Onboarding from '../screens/Onboarding'
+import { pages, carousel } from '../screens/OnboardingPages'
+import PinCreate from '../screens/PinCreate'
+import PinEnter from '../screens/PinEnter'
 import Splash from '../screens/Splash'
+import Terms from '../screens/Terms'
 
-import AuthenticateStack from './AuthenticateStack'
 import ScanStack from './ScanStack'
 import TabStack from './TabStack'
 import defaultStackOptions from './defaultStackOptions'
 
 const Stack = createStackNavigator()
 
-function RootStack() {
-  const [authenticated, setAuthenticated] = useState(false)
+type GenericFn = () => void
+type StateFn = React.Dispatch<React.SetStateAction<boolean>>
 
+const mainStack = (authenticated: boolean, setAuthenticated: StateFn) => {
   return (
-    <Stack.Navigator screenOptions={{ ...defaultStackOptions, headerShown: false }}>
-      {authenticated ? (
-        <Stack.Group>
-          <Stack.Screen name="Tabs">{() => <TabStack />}</Stack.Screen>
-          <Stack.Screen name="Connect" options={{ presentation: 'modal' }}>
-            {() => <ScanStack />}
-          </Stack.Screen>
-        </Stack.Group>
-      ) : (
-        <Stack.Group>
-          <Stack.Screen name="Authenticate" options={{ presentation: 'modal' }}>
-            {() => <AuthenticateStack setAuthenticated={setAuthenticated} />}
-          </Stack.Screen>
-        </Stack.Group>
+    <Stack.Navigator initialRouteName={Screens.Splash} screenOptions={{ ...defaultStackOptions, headerShown: false }}>
+      {!authenticated && (
```

Comment:
Should we still be showing the Tabs and Connect screens if the user is not authenticated?